packages/jsxstyle-utils/src/addStyleToHead.ts

```diff
@@ -23,18 +23,19 @@ if (canUseDOM && !styleElement) {
   document.head.appendChild(styleElement);
 }
 
-export default function addStyleToHead(rule) {
-  if (canUseDOM) {
+export default function addStyleToHead(rule: string) {
+  if (styleElement) {
+    const sheet = styleElement.sheet as CSSStyleSheet;
```

Comment:
Why is the cast needed here?

Reply:
@steadicat the cast feels gross but without it I get `Property 'insertRule' does not exist on type 'StyleSheet'`. Apparently (according to https://github.com/Microsoft/TypeScript/issues/15319) this issue occurs because `styleElement.sheet` might be one of several types.

Reply:
That’s weird, the [type definition of LinkStyle.sheet](https://github.com/Microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L13685) only has one type.

Reply:
Yeah, it’s funky. There’s currently only one type of `StyleSheet` in the spec (`CSSStyleSheet`) but there might be more than one in the future.

https://github.com/Microsoft/TypeScript/issues/15319#issuecomment-296584050
https://drafts.csswg.org/cssom/#the-stylesheet-interface